text/000-rustdoc-jump-to-definition.md

```diff
@@ -0,0 +1,149 @@
+```
+
+It'd require to look for each method in the various files, etc.
+
+So overall, this would greatly improve the source code browsing experience.
+
+The other goal is to allow to go back from the source code pages directly to the item documentation to be able to read the rendered content and not just the "raw" doc comments.
+
+So in short, this feature is split in two parts:
+
+ * Jump to the definition of an ident.
+ * Jump to an item documentation from the source view.
+
+> @GuillaumeGomez comment: I tried to explain why I think having jump to def links and "back to doc" links is important, but I feel like I'm explaining it in some weird and unclear ways... Any suggestion is very welcome.
+
+Adding this information directly into the documentation allows it to improve the documentation value: if you want to go a bit further what is written in the documentation (to see some implementation details for example), you can currently go to the source code. However, you're quickly limited to the current file if it's using private items not from this file.
+
+Once you found what you're looking for, having a link to an item's documentation from the source code allows to add the missing connection between the source code viewer and the doc pages.
+
+Another important note is that if documentation is split in multiple places or requires an external tool for the source code navigation, it's actually slowing down our users quite a lot.
+
+> @GuillaumeGomez comment: This is the "story" part. Is it ok to put it here or should it be somewhere else? Also: is it clear enough?
+
+A good example where this would be very helpful is for the [rust compiler source code](https://doc.rust-lang.org/1.57.0/nightly-rustc/src/rustc_middle/hir/map/mod.rs.html#872-877): even if a fully set up IDE or github, it's very hard to go around without previous knowledge or help from someone else. It's also very common to have undocumented items or to just want to see how something is implemented. With the "jump to definition", it already improved quite a lot the browsing experience by allowing to jump super quickly by simply clicking on a item. The missing part is now to go back from the source code into the item documentation to make it complete.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+When going to the source code pages (as you can see already in the [compiler documentation](https://doc.rust-lang.org/1.57.0/nightly-rustc/src/rustc_middle/hir/map/mod.rs.html#872-877)), you will be able to click on idents to go to their definition.
+
+Once at their definition, if you click on the item name, you'll arrive on its documentation page (if it has one, otherwise nothing will happen).
+
+An interesting thing to note: primitive types always link to documentation. Also, to make it more obvious if a link points to source code or to documentation, they should have different visual markers (different background/underline colors for example).
```

Comment:
The discussion of different visual markers is a bit out of place. In general, considerations of UI are scattered throughout the doc. I think it would be better for most of the doc to focus on the actions people should be able to perform (go to definition; go to documentation), and have a separate section discussion UI possibilities to achieve that.

Reply:
Should I create a new section or should I simply remove it then?

Reply:
I think you should create a new section to discuss UI.

Reply:
Not sure if this has been mentioned already. I would advice against differentiating link types by color since that would harm accessibility for colorblind people (in case of red & blue: monochromacy / achromatopsia, blue cone monochromacy). Just something to keep in mind.

Reply:
We decided to not add new color for links (or background). Instead they will have an underline when hovered.